Port TG's Spirit Boards

nearlyNonexistent opened the pull request:
🆑nearlyNon
rscadd: Non-Branded Spirit Boards, which can talk to ghosts! You need at least 2 people surrounding it to use it. Pick it up with grab intent; otherwise you'll try to speak with ghosts with it. Ghosts interact with it by clicking it. One starts in the Chapel, the other is purchased with a coin in the fun vending machine.
/🆑

nearlyNonexistent commented:
Also someone asked why it starts in the chapel: because it's a religious artifact for the Thelema religion, one of the religions that are available in the selector.

Neon1ight commented:
Honestly, this feels like more of a LRP thing to me. Just feels wrong for our setting and the fact we're an HRP server (A straight up magical board that ghosts can use to talk with people). On top of that it can easily be used to meta, if ghosts can interact with it.

nearlyNonexistent commented:
A fluff object that realistically can't be used to meta, for the fact that literally anyone can add a letter into the queue and disrupt the message, and there's nothing stopping from another ghost from ruining said meta attempt with random letters?

The fact it requires a second human helps balance it out, as the humans will naturally add in random letters too, to prank each other or try to finish sentences/words.

Also, if it's bad for the setting, why is Thelema a religion selectable via the background system?

So far F-Tang Steve raised no objection, nor Kyos though they're retired staff.

nearlyNonexistent commented:
wait I just realized I should move the file to the new path hold on

Nerev4r commented:
You'll have to explain how this is "LRP"

nearlyNonexistent commented:
From discussion on the #feedback channel they said it's LRP because it encourages meta and for ghosts to interact. This has a few balancing systems to prevent it as I mentioned; you have no way of knowing if it was a ghost or one of the other players at the board, or even if it's all from the same ghost.

ghost commented:
Anything that gives the Chaplain more to do is okay in my book.

HeyBanditoz reviewed (changes requested):
You committed a backup map file.

FreePrivateer commented:
I'm gonna agree with LRP here. Being a ghost is a polite way of observing the round, not meant to be any level of IC, for the most part. Its alright if its a religious object, but unless we're going to build some deeper lore as to how /spirits/ work in our sci-fi game, just make it an object. Ghosts should remain out-of-character, meta or not.

nearlyNonexistent commented:
Ghosts already interact with the world. Either remove the ghosts' ability to interact with the world on cult and during bluespace, or this is logically consistent.

And again: ICly there's no way to tell if it truly IS a ghost, or one of the other players messing with you, or your own ideomotor effect. The purpose is to provide a sense of mysticism to the item.

HeyBanditoz commented on code/game/objects/items/spirit_board.dm:
```diff
@@ -0,0 +1,73 @@
+/obj/item/weapon/spirit_board/proc/spirit_board_pick_letter(mob/M)
+	if(!spirit_board_checks(M))
+		return 0
+	planchette = input("Choose the letter.", "Seance!") as null|anything in list(
```
What if a lowercase option is used. Does it still work? Examining should return this list, emphasising it's case sensitive.

HeyBanditoz replied:
Oh wait this isn't a text box it's a list isn't it

nearlyNonexistent replied:
it's a list so you can pick YES/NO/GOODBYE

MistakeNot4892 commented:
If you want 'logically consistent', ghosts should only be able to interact during spooky cult and bluespace. Otherwise you could make it a silent interaction to move the planchette yourself, as ouja boards tend to work IRL.

nearlyNonexistent commented:
buddy that's exactly what's in this PR

MistakeNot4892 commented:
Misunderstood your earlier comment. Haven't reviewed the PR yet.

nearlyNonexistent commented:
I should add a cult check though agreed, but I mean right now moving the planchette is silent already and any human can do it

nearlyNonexistent commented:
there, now it only works on cult mode for all you fun haters ;( otherwise it's just an ordinary board with a planchette people can move and prank people with

nearlyNonexistent commented:
gonna side with other opinions and revert the cult-mode only change, can put it back if required

nearlyNonexistent committed:
Port TGstation's Spirit Board

🆑nearlyNon
rscadd: Non-Branded Spirit Boards, which can talk to ghosts! You need at least 2 people surrounding it to use it. Pick it up with grab intent; otherwise you'll try to speak with ghosts with it. Ghosts interact with it by clicking it. One starts in the Chapel, the other is purchased with a coin in the fun vending machine.
/🆑

babydoll commented:
Mods and retired staff do not make development decisions or review PRs
